### PR TITLE
Add docname & url reordering documentation

### PR DESCRIPTION
**Pull request by gustingonzalez**

### CODE REVIEW

**Comment by amallia** [comment]
Thank you for your PR. I will take a look soon

**Comment by amallia** [comment]
LGTM. Let's wait for @elshize

File: docs/source/document_reordering.md [hunk 1]
```diff
@@ -1,9 +1,51 @@
 # Document Reordering
 
+## Document name & URL based ordering
+To reorder an inverted index based on document names or their URLs, use the `generate_sorted_docids_mapping.py` script and the `shuffle_docids` command.
 
-## Random Ordering
+The first one allows to generate a file in which each line maps a \<current ID\> with a \<new ID\>. These identifiers are generated based on the lexicographical order of  document names (or their URLs). Furthermore, this file serves as input to the `shuffle_docids` command, which materializes the reordering operation.
+
+### 1. Mapping file creation
+To generate the mapping file, it is necessary to take into account that each line number of the `.documents` and `.urls` files (which are part of the forward index) correspond to its docid. So, the first document of `.documents` (line 0, and therefore docid = 0) is equivalent to the first url in the `.urls` file, and so on. In this way, if you want to generate the mapping file to perform a reordering using either files, you should use the following script:
```
Comment (elshize) [inline docs/source/document_reordering.md:9]: "equivalent to the first URL" <--- capitalize

File: docs/source/document_reordering.md [hunk 2]
```diff
@@ -1,9 +1,51 @@
...
+
+```
+usage: script/generate_sorted_docids_mapping.py [-h] documents output
+
+Take a document lexicon (e.g. '.documents' or '.urls' file) and sort it,
```
Comment (elshize) [inline docs/source/document_reordering.md:14]: @amallia By `lexicon`, do we mean the text file, right? I believe we use `lexicon` for the memory mapped payload vector, i.e., `.docmap`, right? Maybe we should rename this?

File: script/generate_sorted_docids_mapping.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+def dump(output, sorted_docids):
+    fout = file(output, "w")
+    for newid in range(0, len(sorted_docids)):
+        # Note that sorted_docids[newid] is the currid.
+        fout.write(str(sorted_docids[newid]) + " " + str(newid) + "\n")
+
+
+def main():
+    parser = ArgumentParser(description='Take a document lexicon (e.g. \'.documents\' or\
```
Comment (elshize) [inline script/generate_sorted_docids_mapping.py:14]: Let's use double quotes for either the string or the quotes inside, and get rid of the escape characters, e.g.,
`"Take a document lexicon (e.g. '.documents' ...`

File: script/generate_sorted_docids_mapping.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                            \'.urls\' file) and sort it, generating a file mapping \'<current ID>\
+                            <new ID>\' to use with the \'suffle_docids\' script.')
+    parser.add_argument('documents', help='File containing one document (or url) per line and \
+                        where each line number (starting from zero) represents its docid')
+    parser.add_argument('output', help='Output file mapping \'<current ID> <new ID>\'')
```
Comment (elshize) [inline script/generate_sorted_docids_mapping.py:19]: Same as above about quotes

File: script/generate_sorted_docids_mapping.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    args = parser.parse_args()
+
+    # Documents dict, where each element maps an id with a document (or url).
+    documents = dict()
+    docid = 0
+    for line in open(args.documents, "r"):
```
Comment (elshize) [inline script/generate_sorted_docids_mapping.py:25]: We don't seem to be using `docid` outside of the loop, so let's use `enumerate`:
```python
for docid, line in enumerate(open(args.documents, "r")):
    documents[docid] = string.lower(line[:-1])
```

Moreover, we can do:

```python
with open(args.documents, "r") as input_file:
    documents = {docid: string.lower(line[:-1]) for docid, line in enumerate(input_file)}
```

**Review by elshize — CHANGES_REQUESTED** [review]

**Comment by elshize** [comment]
@amallia I left some small comments, one for you, take a look.

File: docs/source/document_reordering.md (hunk 2 above)
Comment (amallia) [reply docs/source/document_reordering.md:14]: I would change the sentence to `Take a text file lexicon` to disambiguate it.

File: script/generate_sorted_docids_mapping.py (hunk 3 above)
Comment (amallia) [reply script/generate_sorted_docids_mapping.py:14]: Same as above

**Comment by gustingonzalez** [comment]
> @amallia I left some small comments, one for you, take a look.

After introducing the changes through a force push, the PR has closed. Could it be reopened? Thanks!

**Pull request closed without merging**